### PR TITLE
port changes from LDAPc to LDAPc-ada

### PR DESCRIPTION
**Pull request by mvocu**
Add expiration of group members to LDAPc-ada

### CODE REVIEW

File: perun-ldapc-ada/src/main/resources/perun-ldapc.xml [hunk 1]
```diff
@@ -272,6 +272,36 @@ http://www.springframework.org/schema/aop http://www.springframework.org/schema/
...
+				<bean class="cz.metacentrum.perun.ldapc.processor.impl.RegexpDispatchEventCondition">
+					<property name="beansCondition">
+						<list>
+							<value>cz.metacentrum.perun.core.api.Group</value>
+							<value>cz.metacentrum.perun.core.api.Member</value>
+						</list>
+					</property>
+					<property name="pattern" value="expired.$|disabled.$|invalidated.$|suspended #" />
+					<property name="handlerMethodName" value="processMemberRemoved" />
+				</bean>
```
Comment (stavamichal) [inline perun-ldapc-ada/src/main/resources/perun-ldapc.xml]: These regulars were used in the original configuration, but they are new in this pull request. How did you manage removing members from VO without them before this change?

File: perun-ldapc-ada/src/main/java/cz/metacentrum/perun/ldapc/processor/impl/GroupEventProcessor.java [hunk 2]
```diff
@@ -36,6 +36,18 @@ public void processMemberAdded(String msg, MessageBeans beans) {
...
+			return;
+		}
+		try {
+			log.debug("Removing member {} from group {}", beans.getMember(), beans.getGroup());
+			perunGroup.removeMemberFromGroup(beans.getMember(), beans.getGroup());
+		} catch (NamingException | InternalErrorException e) {
+			log.error("Error removing member {} from group {}: {}", beans.getMember().getId(), beans.getGroup().getId(), e.getMessage());;
+		}
+	}
+
```
Comment (stavamichal) [inline perun-ldapc-ada/src/main/java/cz/metacentrum/perun/ldapc/processor/impl/GroupEventProcessor.java]: This is the first time you need method to remove member from group? So in the basic implementation you are not removing members from group or there is different logic in it? What about "remove Member from Group totally" message?

File: perun-ldapc-ada/src/main/java/cz/metacentrum/perun/ldapc/beans/GroupSynchronizer.java [hunk 3]
```diff
@@ -58,7 +58,7 @@ public void synchronizeGroups() {
 
 							log.debug("Getting list of members for group {}", group.getId());
 							// List<Member> members = ldapcManager.getRpcCaller().call("groupsManager",  "getGroupMembers", params).readList(Member.class);
-							List<Member> members = perun.getGroupsManager().getGroupMembers(ldapcManager.getPerunSession(), group);
+							List<Member> members = perun.getGroupsManager().getActiveGroupMembers(ldapcManager.getPerunSession(), group);
```
[the remark this answers is not in the document]
Comment (mvocu) [reply perun-ldapc-ada/src/main/java/cz/metacentrum/perun/ldapc/beans/GroupSynchronizer.java]: Not sure, you are the core API expert here. However, this interface does not support three parameter version of getActiveGroupMembers(), so I have to change this to *Bl version.

File: perun-ldapc-ada/src/main/resources/perun-ldapc.xml (hunk 1 above)
Comment (mvocu) [reply perun-ldapc-ada/src/main/resources/perun-ldapc.xml]: It is a bug in the older version.

File: perun-ldapc-ada/src/main/java/cz/metacentrum/perun/ldapc/processor/impl/GroupEventProcessor.java (hunk 2 above)
Comment (mvocu) [reply perun-ldapc-ada/src/main/java/cz/metacentrum/perun/ldapc/processor/impl/GroupEventProcessor.java]: Yes to the first two, "totally" was added too.

**Pull request merged**